crates/issue/src/lib.rs

```diff
@@ -465,37 +469,71 @@ impl<T: Config> Pallet<T> {
     fn _cancel_issue(requester: T::AccountId, issue_id: H256) -> Result<(), DispatchError> {
         let issue = Self::get_issue_request_from_id(&issue_id)?;
 
-        // only cancellable after the request has expired
-        ensure!(
-            ext::btc_relay::has_request_expired::<T>(
-                issue.opentime,
-                issue.btc_height,
-                Self::issue_period().max(issue.period)
-            )?,
-            Error::<T>::TimeNotExpired
-        );
-
-        // Decrease to-be-redeemed tokens:
-        let full_amount = issue.amount().checked_add(&issue.fee())?;
-
-        ext::vault_registry::decrease_to_be_issued_tokens::<T>(&issue.vault, &full_amount)?;
-
-        let griefing_collateral = issue.griefing_collateral();
-        if ext::vault_registry::is_vault_liquidated::<T>(&issue.vault)? {
-            griefing_collateral.unlock_on(&issue.requester)?;
-        } else {
+        let griefing_collateral = if ext::btc_relay::has_request_expired::<T>(
+            issue.opentime,
+            issue.btc_height,
+            Self::issue_period().max(issue.period),
+        )? {
+            // anyone can cancel the issue request once expired
+            let griefing_collateral = issue.griefing_collateral();
+            if ext::vault_registry::is_vault_liquidated::<T>(&issue.vault)? {
```

Comment:
Don't we need this check in the `issue.requester == requester` branch below as well? If so, let's refactor it out of the `if` body